src/DotVVM.Framework/Compilation/Javascript/JavascriptTranslatableMethodCollection.cs

```diff
@@ -15,6 +15,25 @@
 
 namespace DotVVM.Framework.Compilation.Javascript
 {
+    public class DelegateInvokeMethodTranslator : IJavascriptMethodTranslator
+    {
+        public JsExpression TryTranslateCall(LazyTranslatedExpression context, LazyTranslatedExpression[] arguments, MethodInfo method)
+        {
+            if (method == null)
+            {
+                return null;
+            }
+
+            if (method.Name == "Invoke" && typeof(Delegate).IsAssignableFrom(method.DeclaringType))
+            {
+                var invocationTargetExpresionCall = context.JsExpression().Invoke(arguments.Select(a => a.JsExpression()));
+                return new JsIdentifierExpression("Promise").Member("resolve").Invoke(invocationTargetExpresionCall)
+                    .WithAnnotation(new ResultIsPromiseAnnotation(a=> a));
```

Comment:
```suggestion
                return invocationTargetExpresionCall
                    .WithAnnotation(new ResultIsPromiseAnnotation(a=>new JsIdentifierExpression("Promise").Member("resolve").Invoke(a)));
```

It would be a bit nicer to wrap it inside promise only when it's going to be waited for.

Reply:
I did not realize it works this way, So the promise accessor is used only if someone wants to chain `.then` behind it? Ok than ;)